src/main/groovy/com/calgaryscientific/gradle/VeracodeBeginScanTask.groovy

```diff
@@ -26,28 +26,78 @@
 
 package com.calgaryscientific.gradle
 
+import groovy.transform.CompileStatic
+
+@CompileStatic
 class VeracodeBeginScanTask extends VeracodeTask {
     static final String NAME = 'veracodeBeginScan'
+    private String app_id
+    private Set<String> moduleWhitelist
 
     VeracodeBeginScanTask() {
-        description = 'Starts a Veracode scan for the application id passed in'
+        description = 'Starts a Veracode scan for given application ID'
         requiredArguments << 'app_id'
+        dependsOn "veracodeGetPreScanResults"
+        app_id = project.findProperty("app_id")
+        defaultOutputFile = new File("${project.buildDir}/veracode", 'begin-scan.xml')
     }
 
-    void run() {
-        def moduleIds = []
-        def whiteList = readListFromFile(new File("src/apps/${project.app_id}/modules-whitelist.txt"))
-        readXml('build/pre-scan-results.xml').each() { module ->
-            if (whiteList.contains(module.@name)) {
-                moduleIds << module.@id
+    VeracodeGetPreScanResultsTask preScan = new VeracodeGetPreScanResultsTask()
+    File preScanResultsOutputFile = preScan.getOutputFile()
+
+    static Set<String> extractWhitelistModuleIds(Node xml, Set<String> whitelist) {
+        Set<String> moduleIds = []
+        Set<String> moduleNames = []
+        NodeList moduleList = xml.getAt("module") as NodeList
+        for (int i = 0; i < moduleList.size(); i++) {
```

Comment:
sounds like this loop is to extract any moduleIds that has status doesn't start with Fatal? then should we extract it to something? extractModulesDoesNotHavePrevisFatal?